src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -76,21 +76,18 @@ ucs_config_field_t uct_ib_mlx5_iface_config_table[] = {
     {NULL}
 };
 
-#if HAVE_DECL_MLX5DV_CQ_INIT_ATTR_MASK_CQE_SIZE
-static ucs_status_t
-uct_ib_mlx5_set_cqe_zipping(uct_ib_mlx5_md_t* md,
-                            struct mlx5dv_cq_init_attr* dv_attr,
-                            const uct_ib_mlx5_iface_config_t* mlx5_config)
+ucs_status_t
+uct_ib_mlx5_parse_cqe_zipping(uct_ib_mlx5_md_t* md,
+                              const uct_ib_mlx5_iface_config_t* mlx5_config,
+                              uct_ib_iface_init_attr_t *init_attr)
 {
-#if HAVE_DECL_MLX5DV_CQ_INIT_ATTR_MASK_COMPRESSED_CQE
     if (mlx5_config->cqe_zipping_enable == UCS_NO) {
         return UCS_OK;
     }
 
-    if (((dv_attr->cqe_size == 64)  && (md->flags & UCT_IB_MLX5_MD_FLAG_CQE64_ZIP)) ||
-        ((dv_attr->cqe_size == 128) && (md->flags & UCT_IB_MLX5_MD_FLAG_CQE128_ZIP))) {
-        dv_attr->comp_mask          |= MLX5DV_CQ_INIT_ATTR_MASK_COMPRESSED_CQE;
-        dv_attr->cqe_comp_res_format = MLX5DV_CQE_RES_FORMAT_CSUM;
+    if ((md->flags & UCT_IB_MLX5_MD_FLAG_CQE64_ZIP) &&
+        (md->flags & UCT_IB_MLX5_MD_FLAG_CQE128_ZIP)) {
```

Comment:
don't we want to check per our cqe size? pass cqe size as param to this function
so user can set cqe=64 if needed to support dev with zipping only on 64, for example